run_unittests.py

```diff
@@ -3290,6 +3290,11 @@ def test_pkgconfig_gen_deps(self):
         out = self._run(cmd + ['--print-requires-private']).strip().split('\n')
         self.assertEqual(sorted(out), sorted(['libexposed', 'libfoo >= 1.0', 'libhello']))
 
+    def check_pkg_flags_are_same(self, output, expected):
+        if is_osx():
```

Comment:
I just tested, haiku fails this test as well, can we add `is_haiku()` here? Then this version looks good, I'll close mine.